.circleci/config.yml

```diff
@@ -0,0 +1,186 @@
+version: 2
+
+defaults: &defaults
+  working_directory: /go/src/github.com/weaveworks/scope
+  docker:
+    - image: weaveworks/scope-backend-build:circle-2.0-4ad2f14
+
+client-defaults: &client-defaults
+  working_directory: /home/weave/scope
+  docker:
+    - image: weaveworks/scope-ui-build:circle-2.0-2885645
+
+workflows:
+  version: 2
+  test_and_deploy:
+    jobs:
+      - lint
+      - unit-test
+      - client-build
+      - client-test:
+          requires:
+            - client-build
+      - xplatform-build:
+          requires:
+            - build
+      - build:
+          requires:
+            - client-build
+      - integration-tests:
+          requires:
+            - lint
+            - unit-test
+            - build
+      - deploy:
+          filters:
+            branches:
+              only: master
+          requires:
+            - client-test
+            - integration-tests
+
+jobs:
+  lint:
+    <<: *defaults
+    steps:
+      - checkout
+      - run: make BUILD_IN_CONTAINER=false lint
+
+  unit-test:
+    <<: *defaults
+    parallelism: 1
+    steps:
+      - checkout
+      - run: COVERDIR=./coverage make BUILD_IN_CONTAINER=false CODECGEN_UID=23 tests
+      - persist_to_workspace:
+          root: .
+          paths:
+          - coverage
+
+  # Create client/build/index.html
+  client-build:
+    <<: *client-defaults
+    steps:
+      - checkout
+      # Convoluted set of steps here to mimic Makefile actions of bind-mounting different dirs into container
+      - run: |
+          mv client/app /home/weave/
+          cd /home/weave; mkdir build         ; yarn run build         ; mv build          scope/client
+          cd /home/weave; mkdir build-external; yarn run build-external; mv build-external scope/client
+      - persist_to_workspace:
+          root: /home/weave/scope
+          paths:
+          - client/build/
+          - client/build-external/
+
+  client-test:
+    <<: *client-defaults
+    steps:
+      - checkout
+      - run: |
+          mv client/app client/test /home/weave/
+          cd /home/weave; yarn run lint
+          cd /home/weave; yarn test
+
+  xplatform-build:
+    <<: *defaults
+    steps:
+      - checkout
+      - run: GOARCH=arm  make BUILD_IN_CONTAINER=false GO_BUILD_INSTALL_DEPS= prog/scope
+      - run: GOOS=darwin make BUILD_IN_CONTAINER=false GO_BUILD_INSTALL_DEPS= prog/scope
```

Comment:
Does this make any artefacts that we need to keep?

Reply:
no